Tests broken in master branch

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

-->

**What happened**:

Tests don't pass, sse: https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/post-external-dns-push-images/1848748383435493376 


**What you expected to happen**:

Tests are green

**How to reproduce it (as minimally and precisely as possible)**:

Run tests in master.

Note: I think there's something funky going on, I can reproduce the tests on one of my machines, but not on another. I'm not exactly sure how can this fail:

```
=== RUN   TestAWSSDProvider_awsTags
    aws_sd_test.go:1007:
        	Error Trace:	/Users/raffo/go/src/sigs.k8s.io/external-dns/provider/awssd/aws_sd_test.go:1007
        	Error:      	Not equal:
        	            	expected: []types.Tag{types.Tag{Key:(*string)(0x14000342d80), Value:(*string)(0x14000342d90), noSmithyDocumentSerde:document.NoSerde{}}, types.Tag{Key:(*string)(0x14000342da0), Value:(*string)(0x14000342db0), noSmithyDocumentSerde:document.NoSerde{}}}
        	            	actual  : []types.Tag{types.Tag{Key:(*string)(0x14000342dc0), Value:(*string)(0x14000342dd0), noSmithyDocumentSerde:document.NoSerde{}}, types.Tag{Key:(*string)(0x14000342de0), Value:(*string)(0x14000342df0), noSmithyDocumentSerde:document.NoSerde{}}}

        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,2 +1,8 @@
        	            	 ([]types.Tag) (len=2) {
        	            	+ (types.Tag) {
        	            	+  Key: (*string)((len=4) "key2"),
        	            	+  Value: (*string)((len=6) "value2"),
        	            	+  noSmithyDocumentSerde: (document.NoSerde) {
        	            	+  }
        	            	+ },
        	            	  (types.Tag) {
        	            	@@ -6,8 +12,2 @@
        	            	   }
        	            	- },
        	            	- (types.Tag) {
        	            	-  Key: (*string)((len=4) "key2"),
        	            	-  Value: (*string)((len=6) "value2"),
        	            	-  noSmithyDocumentSerde: (document.NoSerde) {
        	            	-  }
        	            	  }
        	Test:       	TestAWSSDProvider_awsTags
--- FAIL: TestAWSSDProvider_awsTags (0.00s)
FAIL
FAIL	sigs.k8s.io/external-dns/provider/awssd	1.699s
```

--- a/provider/awssd/aws_sd_test.go
+++ b/provider/awssd/aws_sd_test.go
@@ -1004,6 +1004,6 @@ func TestAWSSDProvider_awsTags(t *testing.T) {
 	}
 
 	for _, test := range tests {
-		assert.EqualValues(t, test.Expectation, awsTags(test.Input))
+		require.ElementsMatch(t, test.Expectation, awsTags(test.Input))
 	}
 }

